distributed/worker.py

```diff
@@ -2369,13 +2369,8 @@ async def plugin_add(self, comm=None, plugin=None, name=None):
         with log_errors(pdb=False):
             if isinstance(plugin, bytes):
                 plugin = pickle.loads(plugin)
-            if not name:
-                if hasattr(plugin, "name"):
-                    name = plugin.name
-                else:
-                    name = funcname(plugin) + "-" + str(uuid.uuid4())
-
-            assert name
+            if name is None:
+                name = get_plugin_name(plugin)
```

Comment:
It looks like the behavior here is to not overwrite the existing plugin.  I wonder if maybe we should overwrite instead?  I'm guessing that this will more often be what the user wants.

Reply:
The behaviour here is the [documented behaviour](https://docs.dask.org/en/latest/futures.html#distributed.Client.register_worker_plugin): "a plugin with that name has already registered then any future plugins will not run". This behaviour was already enforced by the worker, although it is documented for the client.

Perhaps my mistake is in the scheduler, where I do overwrite any existing plugin with the same name.